vapi/simulator/simulator.go

```diff
@@ -967,19 +961,309 @@ func (s *handler) library(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func (content *content) cached(val bool) {
-	for _, item := range content.Item {
-		item.cached(val)
+func (s *handler) syncSubLib(dstLib *content) error {
+
+	sub := dstLib.Subscription
+	if sub == nil {
+		return nil
+	}
+
+	lastSyncTime := time.Now().UTC()
+	dstLib.LastSyncTime = &lastSyncTime
+
+	var syncAll bool
+	if sub.OnDemand != nil && !*sub.OnDemand {
```

Comment:
Just to comment that OnDemand is required on the API, should never be nil.

Reply:
Fair, but since it is a nillable field, it's best to check this anyway. We can do a follow-up to make GoVmomi treat it as a non-nillable field.